### PR TITLE
refactor: remove interior mutability from TrackerRegistry

### PR DESCRIPTION
**Pull request by tustvold**
We want to be able to reclaim from the TrackerRegistry and enqueue the removed data in an archive as one atomic operation. This means doing the reclaim and re-enqueue under a single lock, otherwise an external observer could potentially see missing or duplicated jobs, depending on how reads to the archive and registry are sequenced.

Nothing about the TrackerRegistry having an internal mutex prevents this, but getting rid of it removes what would then be a largely redundant mutex, and also allows a more efficient "streaming" reclaim without having to buffer via a Vec.

### CODE REVIEW

File: server/src/lib.rs [hunk 1]
```diff
@@ -419,12 +420,15 @@ impl<M: ConnectionManager> Server<M> {
...
             // Dequeue
-            let reclaimed = self.jobs.reclaim();
+            let mut jobs = self.jobs.lock();
 
-            for job in reclaimed {
+            for job in jobs.reclaim() {
                 info!(?job, "job finished");
             }
 
+            // Ensure mutex guard is not held across await point
```
Comment (tustvold) [inline server/src/lib.rs:429]: As the future is currently run directly on the main function, it doesn't require the future to have Send bounds, which means this isn't a compile error.

Comment (alamb) [reply server/src/lib.rs:429]: Another way to structure the code to make it harder to accidentally hold the mutex across `await` might be to put the work in a separate function so the loop looks like

```
loop {
  self.reclaim_finished_jobs(); // mutex is taken / released in here
  interval.tick().await
}
```

**Review by alamb — APPROVED** [review]
makes sense to me

**Review by domodwyer — APPROVED** [review]

**Comment by alamb** [comment]
Note this is part of #942

File: server/src/lib.rs (hunk 1 above)
Comment (tustvold) [reply server/src/lib.rs:429]: Will do this in #1005 as it just becomes a single function call then

**Pull request merged**